File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int zeroCount = 0;
        int product = 1;
        for (int num : nums) {
            if (num) {
                product *= num;
            } else zeroCount++;
        }

        for (int i = 0; i < nums.size(); ++i) {
            if (nums[i] == 0) {
                zeroCount--;
                if (zeroCount) {
                    nums[i] = 0;
                } else nums[i] = product;
                zeroCount++;
            } else {
                if (zeroCount) {
                    nums[i] = 0;
                } else nums[i] = product / nums[i];
            }
        }
        return nums;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        contains = set(nums)
        longest = 0
        for num in nums:
            if num - 1 in contains:
                continue
            curr = 1
            while num + curr in contains:
                curr += 1
            longest = max(longest, curr)

        return longest